.gitlab-ci.yml

```diff
@@ -21,84 +21,112 @@ generate-scripts:
       - install_script_agent6.sh
       - install_script_agent7.sh
 
-test-opensuse-13:
+.test:
   image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/docker_x64:$DATADOG_AGENT_BUILDIMAGES
   tags: ["runner:docker"]
   stage: test
   dependencies: ["generate-scripts"]
+  script:
+    - ./test/dockertest.sh --image "registry.ddbuild.io/images/mirror/${IMAGE}" --script install_script_agent6.sh 
+      --minor-version "${MINOR_VERSION}"
+      --expected-minor-version "${EXPECTED_MINOR_VERSION}" 
+      --flavor "${FLAVOR}" 
+      --old-suse "${OLD_SUSE}" 
+      --apt-url "$APT_URL" 
+      --apt-repo-version "$APT_REPO_VERSION_AGENT6" 
+      --yum-url "$YUM_URL" 
+      --yum-version-path "$YUM_VERSION_PATH_AGENT6" 
+    - ./test/dockertest.sh --image "registry.ddbuild.io/images/mirror/${IMAGE}" --script install_script_agent7.sh
+      --minor-version "${MINOR_VERSION}"
+      --expected-minor-version "${EXPECTED_MINOR_VERSION}"
+      --flavor "${FLAVOR}"
+      --old-suse "${OLD_SUSE}"
+      --apt-url "$APT_URL"
+      --apt-repo-version "$APT_REPO_VERSION_AGENT7"
+      --yum-url "$YUM_URL"
+      --yum-version-path "$YUM_VERSION_PATH_AGENT7"
+  variables:
+    OLD_SUSE: "false"
+
+test-opensuse-13:
+  extends: .test
+  rules:
+    - if: '$CI_PIPELINE_SOURCE == "push"'
   parallel:
     matrix:
       - IMAGE: opensuse/archive:13.2
       - IMAGE: opensuse/archive:13.2
         FLAVOR: datadog-dogstatsd
       - IMAGE: opensuse/archive:13.2
         FLAVOR: datadog-iot-agent
-  script:
-    - ./test/dockertest.sh --image "registry.ddbuild.io/images/mirror/${IMAGE}" --script install_script_agent6.sh --minor-version "${MINOR_VERSION}" --expected-minor-version "${EXPECTED_MINOR_VERSION}" --flavor "${FLAVOR}" --old-suse "true"
-    - ./test/dockertest.sh --image "registry.ddbuild.io/images/mirror/${IMAGE}" --script install_script_agent7.sh --minor-version "${MINOR_VERSION}" --expected-minor-version "${EXPECTED_MINOR_VERSION}" --flavor "${FLAVOR}" --old-suse "true"
+  variables:
+    OLD_SUSE: "true"
 
+test_pinned_version:
```

Comment:
question ❓ 
As you created this one, why not creating another one for `test_flavours`?
If all test start at same time having several matrixes or only one is basically the same so I guess your objective was only readability? Then we could set these appart too, wdyt?

Reply:
The objective here was not really readbility but rather, being able to not trigger install-script test with pinned versions when the pipeline is triggered from another pipeline. 
Because when we trigger it from an external pipeline we want to run the test with the version corresponding to the pipeline that triggered it.
That's why we have 
```
  rules:
    - if: '$CI_PIPELINE_SOURCE == "push"'
 ```
In this job and not in the other with al the other test cases